instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/instrumentation/kafka/ConcurrentKafkaListenerContainerFactoryPostProcessor.java

```diff
@@ -29,7 +33,12 @@ public Object postProcessAfterInitialization(Object bean, String beanName) {
     ConcurrentKafkaListenerContainerFactory<?, ?> listenerContainerFactory =
         (ConcurrentKafkaListenerContainerFactory<?, ?>) bean;
     SpringKafkaTelemetry springKafkaTelemetry =
-        SpringKafkaTelemetry.create(openTelemetryProvider.getObject());
+        SpringKafkaTelemetry.builder(openTelemetryProvider.getObject())
+            .setCaptureExperimentalSpanAttributes(
+                configPropertiesProvider
+                    .getObject()
+                    .getBoolean("otel.instrumentation.kafka.experimental-span-attributes", false))
```

Comment:
why not use regular spring configuration?

Reply:
- issues with older spring versions 
- support for properties suppliers

Reply:
Perhaps it would be worth to add a comment in the code to explain the reasons of the current implementation

Reply:
what would be a good place?